facebook/ThreatExchange
# Convert Descriptor ID to Indicator ID

Summary
---------

We were in correctly storing the indicator ID as the banked content ID when fetching from ThreatUpdates.

1. Change simple serialization in `python-threatexchange` to store both indicator id and descriptor roll up in CSV output. (will need to bump pip version in later PR) This effects how csv's are stored in s3
2. Change the post apply function in the fetcher which reads csv rows and creates PDQMetadata objects to now store the indicator ID rather thann the descriptor ID
3. Change the s3_adapters to read the new column names from the CSV
4. Change the ThreatExchange writebacker to recognize that `id` now refers to indicator id not descriptor ID 

NOTE: After this lands terraform destroy is probably needed

Test Plan
---------
Retested after rebase

1. Make upload_docker and update lambdas
2. delete all data in s3
3. refetch ThreatExchange data in s3 and confirm the csv columns are correct
4. Watch as index is rebuilt
5. Upload image using HMA UI
6. Watch match, copy ID, and using ID tool confirm that the ID is the correct indicator ID:
<img width="1680" alt="Screen Shot 2021-05-06 at 9 27 41 AM" src="https://user-images.githubusercontent.com/24800630/117332982-51061500-ae4d-11eb-9891-6c805d65d9bb.png">
